packages/storybook/src/stories/grid-pagination.stories.tsx

```diff
@@ -199,10 +210,104 @@ export function AutoPagination() {
           columns={data.columns}
           options={{
             pagination: true,
-            paginationAutoPageSize: true,
+            autoPageSize: true,
           }}
         />
       </div>
     </React.Fragment>
   );
 }
+
+function loadServerRows(params: PageChangedParams): Promise<GridData> {
+  return new Promise<GridData>((resolve) => {
+    getData(params.pageSize, 10).then((data) => {
+      setTimeout(() => {
+        const minId = (params.page - 1) * params.pageSize;
+        data.rows.forEach((row) => {
+          row.id = (Number(row.id) + minId).toString();
+        });
+        resolve(data);
+      }, 500);
+    });
+  });
+}
+
+export function ServerPaginationWithApi() {
+  const apiRef: ApiRef = useApiRef();
+  const data = useData(100, 10);
+  const [rows, setRows] = useState<RowsProp>([]);
+  const [isLoading, setLoading] = useState<boolean>(false);
+
+  React.useEffect(() => {
+    let unsubscribe;
+    if (apiRef && apiRef.current) {
+      unsubscribe = apiRef.current.onPageChanged((params) => {
+        action('onPageChanged')(params);
+        setLoading(true);
+        loadServerRows(params).then((newData) => {
+          setRows(newData.rows);
```

Comment:
Is this story meant to be converted into a demo? If we do, we need to account for the case where the request resolves after the component is unmounted, we also need to replace the relative import to `getData` to one from a public npm package.

Reply:
No, it's not, demos are in the demos folder in documentation

Reply:
Should still be guarded. Otherwise older requests might override newer requests.

Reply:
> No, it's not, demos are in the demos folder in documentation

Ok so it's only for internal quick and dirty prototyping.

Reply:
Yes it's a story to simulate a server side call and how you can do pagination server side